Jenkinsfile

```diff
@@ -1,6 +1,77 @@
+/**
+ * Build a docker image given the name
+ *
+ * @param image_name the name of the image
+ * @return image the built docker image
+ */
+def build_image(image_name) {
+    def image
+    withDockerServer([uri: PCIC_DOCKER]) {
+        image = docker.build(image_name)
+    }
+
+    return image
+}
+
+
+/**
+ * If the master branch has been tagged we also add the `latest` tag.  Otherwise
+ * we just use the branch name as the tag.
+ *
+ * @return tags a list of the tags for the image
+ */
+def get_tags() {
+    String tag = sh (script: 'git tag --contains', returnStdout: true).trim()
+
+    def tags = []
+    if(BRANCH_NAME == 'master' && !tag.isEmpty()) {
+        tags.add(tag)
+        tags.add('latest')
+    } else {
+        tags.add(BRANCH_NAME)
+    }
+
+    return tags
+}
+
+
+/**
+ * Given an image publish it with a tag to the PCIC docker registry.
+ *
+ * @param image to publish
+ * @return tag to use later in the security scan
+ */
+def publish_image(image) {
+    def tags = get_tags()
+
+    withDockerServer([uri: PCIC_DOCKER]){
+        docker.withRegistry('', 'PCIC_DOCKERHUB_CREDS') {
+            tags.each { tag ->
+                image.push(tag)
+            }
+        }
+    }
+
+    return tags[0]
```

Comment:
Why do you only return the first?

Reply:
So in the security scan I just want to scan the image that was just pushed. To identify it I just need one of the tags, so arbitrarily I choose the first one. I updated the function to return the full list and then just use the first item later on.